.github/workflows/test.yml

```diff
@@ -39,13 +44,23 @@ jobs:
           key: ${{ runner.os }}-tsc-${{ hashFiles('**/yarn.lock') }}-${{ github.run_id }}
           restore-keys: ${{ runner.os }}-tsc-${{ hashFiles('**/yarn.lock') }}-
       - run: yarn typecheck
+      - if: failure() && github.ref_name == 'main'
+        uses: ./.github/actions/report
+        with:
+          name: Typecheck
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_TEST_REPORTER_WEBHOOK }}
 
   deps-tests:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
       - uses: ./.github/actions/setup
       - run: yarn yarn-deduplicate --strategy=highest --list --fail
+      - if: failure() && github.ref_name == 'main'
+        uses: ./.github/actions/report
+        with:
+          name: Dependency checks
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_TEST_REPORTER_WEBHOOK }}
```

Comment:
failure here would not cause runtime issues, so I don't think it warrants an alert

Reply:
that's true, but i guess i would personally just like to know about any red x's on main - and my understanding is this just pings a slack channel, not triggers pagerduty right?

Reply:
It would _likely_ not cause runtime issues, but it might cause some version skew bugs, and it _would_ prevent new PRs from being merged, as `deps-tests` is required to pass per our branch protections.